packages/smart-wallet/src/smartWallet.js

```diff
@@ -467,16 +429,14 @@ export const prepareSmartWallet = (baggage, shared) => {
                * @returns {Promise<RemotePurse>}
                */
               purseForBrand: async brand => {
-                if (brandPurses.has(brand)) {
-                  return brandPurses.get(brand);
+                if (registry.has(brand)) {
+                  // @ts-expect-error RemotePurse cast
+                  return E(bank).getPurse(brand);
+                } else if (invitationBrand === brand) {
+                  // @ts-expect-error RemotePurse cast
+                  return invitationPurse;
                 }
-                const desc = registry.getRegisteredAsset(brand);
-                const { bank } = this.state;
-                /** @type {RemotePurse} */
-                // @ts-expect-error cast to RemotePurse
-                const purse = E(bank).getPurse(desc.brand);
-                await facets.helper.addBrand(desc, purse);
-                return purse;
+                throw assert.error(`cannot find/make purse for ${brand}`);
```

Comment:
Is `throw` necessary for `never`?
```suggestion
                throw Fail`cannot find/make purse for ${brand}`;
```

Reply:
`assert.error` is not never; it constructs an error. But `Fail` is fine

Reply:
yeah I was asking why the `throw`. because without it `assert.fail` is an option.

Reply:
Something about "arrow function needs return" IIRC